analyses/ZplusJet_3.cc

```diff
@@ -107,103 +132,147 @@ namespace Rivet {
 
       // Retrieve dressed leptons, sorted by pT
       vector<DressedLepton> leptons = apply<DressedLeptons>(event, "leptons").dressedLeptons();
-      MSG_DEBUG("Lepton multiplicity = " << leptons.size());
 
-      // discard events with less than two muons
+      // discard events with less than two and more than maximum number of leptons
       if (leptons.size() < 2) vetoEvent;
+      if (leptons.size() > _maxnleptons) vetoEvent;
+      MSG_DEBUG("Found " << leptons.size() << " leptons");
+      for (auto lep: leptons) {
+        MSG_DEBUG("\tlepton pt: " << lep.pT());
+        MSG_DEBUG("\tlepton y: " << lep.rap());
+      }
```

Comment:
In der analyse wird der nMuon cut erst angewendet, nachdem die validMuons mit dem pt und eta cut gemacht wurden.

Reply:
Furthermore, in the analysis leptons have a cut on eta, not rapidity

Reply:
That's how it is also done in the analysis. The `pt` and `eta` cuts on the leptons are already applied when creating the collection in the `init` method:
https://github.com/HerrHorizontal/herwig-run/blob/update/rivet-ana/analyses/ZplusJet_3.cc#L64-L66